renderdoc/driver/gl/wrappers/gl_buffer_funcs.cpp

```diff
@@ -608,11 +608,18 @@ void WrappedOpenGL::glNamedBufferDataEXT(GLuint buffer, GLsizeiptr size, const v
       record->Map.orphaned = true;
   }
 
+  GLResourceRecord *record = GetResourceManager()->GetResourceRecord(BufferRes(GetCtx(), buffer));
```

Comment:
I'd rather have the record fetched only inside the `if(IsBackgroundCapturing(m_State))` branch since records are not present on replay so it'll be NULL. Having the variable only in scope inside the capture branch makes it less likely to hit an accidental crash, even if it does the fetch multiple times for multiple branches (there's already a second one above).

Reply:
OK, sounds good.

Reply:
Currently in most of those cases there is an assert inside generic `IsCaptureMode`, that a record should present. But I think it's still a good idea to have it local to the `IsBackgroundCapture` branch, since it makes any future refactoring easier.